src/core_modules/capture-core/components/Pages/MainPage/WorkingLists/ListViewLoader/useIsContextInSync.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+import { useMemo } from 'react';
+import isSelectionsEqual from '../../../../App/isSelectionsEqual';
+import type { Categories } from '../workingLists.types';
+
+export const useIsContextInSync = (
+    programId: string,
+    orgUnitId: string,
+    categories?: Categories,
+    viewContext: ?Object,
+) => useMemo(() => {
+    if (!viewContext) {
+        return false;
+    }
+
+    const currentSelections = {
+        programId,
+        orgUnitId,
+        categories,
+    };
+
+    return isSelectionsEqual(currentSelections, viewContext);
+}, [
+    programId,
+    orgUnitId,
+    categories,
+    viewContext,
+]);
```

Comment:
seeing this in a different file though it makes me think that this is somehow different from all the other hooks.

Is it a lot to ask to move it to the file its being used if we arent reusing it?

Reply:
I would like to keep this separate. Hope we can agree this is not a blocker and move on.

Reply:
Apologize for my snappiness. I just don’t like spending time discussing these minor subjective details. My reasoning for keeping it separate (and not spending time changing it): I can see that most(all) of our custom hooks that are not reused are currently placed in the same file as the component (as you mention). This is fine with me as long as the file doesn’t grow too large, but separating them into their own file is also ok and could even improve readability for me regardless of file size. For this particular hook that I did extract, I feel it is not bound to the component in the same way as the other hooks and it was good to get it out of the way (it is reusable, but since it is put in the same directory as the component it is quite clear to me that it is not being reused at this time).

End of the day, I don't feel there is any value in changing it.